src/shims/unix/android/thread.rs

```diff
@@ -0,0 +1,69 @@
+use rustc_span::Symbol;
+use rustc_target::spec::abi::Abi;
+
+use crate::helpers::check_min_arg_count;
+use crate::shims::unix::thread::{DEFAULT_THREAD_NAME, EvalContextExt as _};
+use crate::*;
+
+pub fn prctl<'tcx>(
+    this: &mut MiriInterpCx<'tcx>,
+    link_name: Symbol,
+    abi: Abi,
+    args: &[OpTy<'tcx>],
+    dest: &MPlaceTy<'tcx>,
+) -> InterpResult<'tcx> {
+    // We do not use `check_shim` here because `prctl` is variadic. The argument
+    // count is checked bellow.
+    this.check_abi_and_shim_symbol_clash(abi, Abi::C { unwind: false }, link_name)?;
+
+    // FIXME: Use PR_SET_NAME and PR_GET_NAME constants when
+    // https://github.com/rust-lang/libc/pull/3941 lands.
+    const PR_SET_NAME: i32 = 15;
+    const PR_GET_NAME: i32 = 16;
+    const TASK_COMM_LEN: usize = 16;
```

Comment:
Seems like the logic would be nicer if you make this `16 - 1`, so that this is really the length of the name.
```suggestion
    const MAX_NAME_LEN: usize = 16-1; // 16 *including* the null terminator
```